shm/CMakeLists.txt

```
@@ -1,5 +1,9 @@
-project(SHM)
+cmake_minimum_required(VERSION 3.10.0)
 
-add_executable(${PROJECT_NAME} main.cpp ship.cpp cargo.cpp island.cpp)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
```

instead of add_compile_options
```suggestion
set(CMAKE_CXX_FLAGS "-g -O0 -Wall -Wextra -Wconversion -pedantic -Werror")
```